solidity/random-beacon/contracts/RandomBeacon.sol

```diff
@@ -771,6 +787,37 @@ contract RandomBeacon is Ownable {
         }
     }
 
+    /// @notice Reports unauthorized groups signing. Must provide a valid signature
+    ///         of the sender's address as a message. Successful signature
+    ///         verification means the private key has been leaked and all group
+    ///         members should be punished by slashing their tokens. Group has
+    ///         to be active or expired. Unauthorized signing cannot be reported
+    ///         for a terminated group. In case of reporting unauthorized
+    ///         signing for a terminated group, or when the signature is invalid,
+    ///         function reverts.
+    /// @param signedMsgSender Signature of the sender's address as a message.
+    /// @param groupId Group that is being reported for leaking a private key.
+    function reportUnauthorizedSigning(
+        bytes memory signedMsgSender,
+        uint64 groupId
+    ) external {
+        Groups.Group memory group = groups.getGroup(groupId);
+
+        relay.validateUnauthorizedSigning(signedMsgSender, group);
+
+        groups.terminateGroup(groupId);
+
+        address[] memory groupMembers = sortitionPool.getIDOperators(
+            groups.getGroup(groupId).members
+        );
+        staking.seize(
+            authorization.minimumAuthorization,
+            unauthorizedSigningNotificationRewardMultiplier,
+            msg.sender,
+            groupMembers
+        );
```

Comment:
We should emit an event informing why tokens have been slashed. See `DkgMaliciousResultSlashed` and `RelayEntryTimeoutSlashed`.